packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -68,56 +75,88 @@ const sortByDisabledThenExpiryDate = (a: BatchRow, b: BatchRow) => {
   return 0;
 };
 
+const useStockLines = (itemCode: string | undefined) => {
+  const [batchRows, setBatchRows] = useState<BatchRow[]>([]);
+  const { data, isLoading } = useItems({
+    code: { equalTo: itemCode },
+  });
+
+  useEffect(() => {
+    if (!itemCode) return;
+    data?.onFilterByCode(itemCode);
+  }, [itemCode]);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const { nodes } = data;
+    const { availableBatches } = nodes[0] ?? { availableBatches: [] };
+
+    setBatchRows(() => {
+      return availableBatches
+        .map(batch => ({
+          ...batch,
+          quantity: 0,
+        }))
+        .sort(sortByDisabledThenExpiryDate);
+    });
+  }, [data]);
+
+  const onIssue = (stockLineId: string, quantity: number) => {
+    const batchRowIdx = batchRows.findIndex(batch => batch.id === stockLineId);
+    const batchRow = batchRows[batchRowIdx];
+    if (!batchRow) return;
+
+    batchRows[batchRowIdx] = { ...batchRow, quantity };
+    setBatchRows([...batchRows]);
+  };
+
+  return { batchRows, onIssue, isLoading };
+};
+
```

Comment:
maybe a bad name..

Just querying for the stock lines and then doing the transform into batch rows here.

Need to make a change to take in to account the invoice lines which already exist in the invoice when setting the quantity.